aws_local/src/lib.rs

```diff
@@ -0,0 +1,159 @@
+use anyhow::{anyhow, Result};
+use aws_config::meta::region::RegionProviderChain;
+use aws_sdk_s3::{Client, Endpoint, Region};
+use chrono::Utc;
+use file_store::traits::MsgBytes;
+use file_store::{file_sink, file_upload, FileStore, FileType, Settings};
+use std::path::Path;
+use std::{str::FromStr, sync::Arc};
+use tokio::sync::Mutex;
+use tonic::transport::Uri;
+use uuid::Uuid;
+
+pub const AWSLOCAL_DEFAULT_ENDPOINT: &str = "http://127.0.0.1:4566";
+
+pub fn gen_bucket_name() -> String {
+    format!("mvr-{}-{}", Uuid::new_v4(), Utc::now().timestamp_millis())
+}
+
+// Interacts with the locastack.
+// Used to create mocked aws buckets and files.
+pub struct AwsLocal {
+    pub fs_settings: Settings,
+    pub file_store: FileStore,
+    pub aws_client: aws_sdk_s3::Client,
+}
+
+impl AwsLocal {
+    async fn create_aws_client(settings: &Settings) -> aws_sdk_s3::Client {
+        let endpoint: Option<Endpoint> = match &settings.endpoint {
+            Some(endpoint) => Uri::from_str(endpoint)
+                .map(Endpoint::immutable)
+                .map(Some)
+                .unwrap(),
+            _ => None,
+        };
+        let region = Region::new(settings.region.clone());
+        let region_provider = RegionProviderChain::first_try(region).or_default_provider();
+
+        let mut config = aws_config::from_env().region(region_provider);
+        config = config.endpoint_resolver(endpoint.unwrap());
+
+        let creds = aws_types::credentials::Credentials::from_keys(
+            settings.access_key_id.as_ref().unwrap(),
+            settings.secret_access_key.as_ref().unwrap(),
+            None,
+        );
+        config = config.credentials_provider(creds);
+
+        let config = config.load().await;
+
+        Client::new(&config)
+    }
+
+    pub async fn new(endpoint: &str, bucket: &str) -> AwsLocal {
+        let settings = Settings {
+            bucket: bucket.into(),
+            endpoint: Some(endpoint.into()),
+            region: "us-east-1".into(),
+            access_key_id: Some("random".into()),
+            secret_access_key: Some("random2".into()),
+        };
+        let client = Self::create_aws_client(&settings).await;
+        client.create_bucket().bucket(bucket).send().await.unwrap();
+        AwsLocal {
+            aws_client: client,
+            fs_settings: settings.clone(),
+            file_store: file_store::FileStore::from_settings(&settings)
+                .await
+                .unwrap(),
+        }
+    }
+    pub async fn put_proto_to_aws<T: prost::Message + MsgBytes>(
+        &self,
+        items: Vec<T>,
+        file_type: FileType,
+        metric_name: &'static str,
+    ) -> Result<String> {
+        // Uuid uses as random to avoid colisions
+        let uuid: Uuid = Uuid::new_v4();
+        let dir_path = format!("/tmp/{}/{}", uuid, self.fs_settings.bucket);
```

Comment:
Can we use `tempdir` here as well?
We don't need to do any cleanup ourselves. All files are removed when the TempDir is dropped.

In the PR we first encountered `AwsLocal` (https://github.com/novalabsxyz/mobile-rewards-estimator/pull/20), this function was never used. Is it in fact not used in any of your projects and we can drop it?

Reply:
@michaeldjeffrey 
This function is used at least in the CAS. It serves as the basis for functions that want to put a particular type of proto file.

https://github.com/novalabsxyz/mobile-coverage/blob/master/cas/tests/common/aws_mock.rs#L141

> Can we use tempdir here as well?
Yes, changed